cns/kubecontroller/nodenetworkconfig/conversion_linux.go

```diff
@@ -48,6 +46,7 @@ func createNCRequestFromStaticNCHelper(nc v1alpha.NetworkContainer, primaryIPPre
 	}
 
 	return &cns.CreateNetworkContainerRequest{
+		HostPrimaryIP:        nc.NodeIP,
```

Comment:
Based on offline discussion with @thatmattlong : "anything not in pod cidr would get snatted with node IP in overlay so there is no need to do an explicit snat for IMDS "